src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -121,6 +121,8 @@ public abstract partial class Compilation
         /// </summary>
         public abstract string Language { get; }
 
+        internal abstract ImmutableDictionary<string, string> SerializeForPdb();
```

Comment:
The name of this method is ambiguous. What is being serialized here? That should be included in the name.

Reply:
I'd call it `SerializePdbEmbeddedCompilationOptions` and also instead of returning dictionary this method can take `BlobBuilder` and write the options directly.

---
In reply to: [427421516](https://github.com/dotnet/roslyn/pull/44373#discussion_r427421516) [](ancestors = 427421516)